packages/http/src/mocker/HttpMocker.ts

```diff
@@ -33,24 +33,36 @@ class HttpMocker
     const payloadGenerator: PayloadGenerator =
       config && typeof config.mock !== 'boolean' && config.mock.dynamic ? generate : generateStatic;
 
-    return pipe(
-      withLogger(logger => {
-        // setting default values
-        const acceptMediaType = input.data.headers && caseless(input.data.headers).get('accept');
-        config = config || { mock: false, validateRequest: true, validateResponse: true };
-        const mockConfig: IHttpOperationConfig =
-          config.mock === false ? { dynamic: false } : Object.assign({}, config.mock);
-
-        if (!mockConfig.mediaTypes && acceptMediaType) {
-          logger.info(`Request contains an accept header: ${acceptMediaType}`);
-          mockConfig.mediaTypes = acceptMediaType.split(',');
-        }
-
-        return mockConfig;
-      }),
-      chain(mockConfig => negotiateResponse(mockConfig, input, resource)),
-      chain(result => assembleResponse(result, payloadGenerator)),
-    );
+    const isHead = input.data.method === 'head';
+    const r = get(resource, ['responses', '0'], { code: 200, headers: [] });
+
+    return isHead
+      ? withLogger(logger => {
+          logger.info('Returning an empty response for a HEAD request.');
+
+          return right({
+            statusCode: +r.code,
```

Comment:
Aha! Nice trick!

Reply:
What does this do? Is there a more readable equivalent?

Reply:
@philsturgeon, if you know JS, then this is readable as readable gets

Reply:
@lag-of-death I think we can use `parseInt` as well — it _will_ effectively improve the readability, right?

Reply:
@XVincentX, don't know, the same for me. Is your preference to change it to `parseInt`?

Reply:
I'm fine with both, I was just pointing out there's an alternative that (maybe) is more readable.

Reply:
This was a long conversation @lag-of-death, please can you be aware and respectful of peoples time. I asked if there was a more readable alternative and you could have said "Yes" and made the code more readable. If I was asking, somebody else will also appreciate it. 

As a general rule, readable is usually preferable to clever. 🙌